src/auxinfo/participant.rs

```diff
@@ -588,14 +608,18 @@ mod tests {
         }
 
         pub fn is_auxinfo_done(&self, auxinfo_identifier: Identifier) -> Result<bool> {
-            let mut fetch = vec![];
-            for participant in self.other_participant_ids.clone() {
-                fetch.push((StorageType::Public, auxinfo_identifier, participant));
-            }
-            fetch.push((StorageType::Public, auxinfo_identifier, self.id));
-            fetch.push((StorageType::Private, auxinfo_identifier, self.id));
-
-            self.storage.contains_batch(&fetch)
+            Ok(self
+                .local_storage
+                .contains_for_all_ids::<PublicStorageType>(
+                    auxinfo_identifier,
+                    &self.other_participant_ids,
+                )
+                && self
+                    .local_storage
+                    .contains::<PublicStorageType>(auxinfo_identifier, self.id)
```

Comment:
suggestion: you can consolidate these on rebase with the `self.all_participants()` method, which combines `other_participant_ids` and `self.id` into one list.

This looks way nicer though!